### PR TITLE
Integrate Jakarta Mail API 2.1.0

### PR DESCRIPTION
**Pull request by mabartos**

### CODE REVIEW

**Comment by vmuzikar** [comment]
@mabartos Can you please rebase?

**Comment by mabartos** [comment]
Started fail Base IT (3); I'm not sure why. I need to investigate.

**Comment by mabartos** [comment]
@vmuzikar I've double-checked the failures; nothing is related to this PR. It works as expected. I also did some manual testing.

File: pom.xml [hunk 1]
```diff
@@ -424,9 +425,14 @@
...
-                <version>${com.sun.jakarta.mail.version}</version>
+                <groupId>jakarta.mail</groupId>
+                <artifactId>jakarta.mail-api</artifactId>
+                <version>${jakarta.mail.version}</version>
+            </dependency>
+            <dependency>
+                <groupId>org.eclipse.angus</groupId>
+                <artifactId>angus-mail</artifactId>
+                <version>${angus.mail.version}</version>
             </dependency>
```
Comment (vmuzikar) [inline pom.xml:436]: If I understand correctly, this dependency will be used just in tests, correct?

And just a silly question but why do we need it at all now? It is a replacement for what dependency?

Comment (mabartos) [reply pom.xml]: Thanks, Vasek, for the review. I've just rechecked the dependencies and made some minor changes.

File: pom.xml [hunk 2]
```diff
@@ -85,8 +85,9 @@
         <jackson.annotations.version>${jackson.version}</jackson.annotations.version>
 
         <!--JAKARTA-->
-        <com.sun.jakarta.mail.version>2.0.1</com.sun.jakarta.mail.version>
         <jakarta.persistence.version>3.1.0</jakarta.persistence.version>
+        <jakarta.mail.version>2.1.1</jakarta.mail.version>
+        <angus.mail.version>2.0.1</angus.mail.version>
```
Comment (vmuzikar) [inline pom.xml:90]: Do we still need `angus.mail.version` property?

File: saml-core-api/pom.xml [hunk 3]
```diff
@@ -64,6 +64,12 @@
         <dependency>
             <groupId>com.sun.xml.ws</groupId>
             <artifactId>rt</artifactId>
+            <exclusions>
+                <exclusion>
+                    <groupId>org.eclipse.angus</groupId>
+                    <artifactId>angus-activation</artifactId>
+                </exclusion>
+            </exclusions>
```
Comment (vmuzikar) [inline saml-core-api/pom.xml:72]: After the recent changes to this PR, do we still need this?

File: testsuite/utils/pom.xml [hunk 4]
```diff
@@ -109,6 +109,10 @@
                     <groupId>org.slf4j</groupId>
                     <artifactId>slf4j-simple</artifactId>
                 </exclusion>
+                <exclusion>
+                    <groupId>org.eclipse.angus</groupId>
+                    <artifactId>angus-activation</artifactId>
+                </exclusion>
```
Comment (vmuzikar) [inline testsuite/utils/pom.xml:115]: After the recent changes to this PR, do we still need this?

File: pom.xml (hunk 2 above)
Comment (mabartos) [reply pom.xml:90]: Current state: Yes, we do, as we need to provide an implementation for the Jakarta Mail API.

File: saml-core-api/pom.xml (hunk 3 above)
Comment (mabartos) [reply saml-core-api/pom.xml]: AFAIK, some SAML tests failed previously, due to the conflict between the Jakarta Activation and Angus activation (probably). I'll check the results for this GHAs run.

File: testsuite/utils/pom.xml (hunk 4 above)
Comment (mabartos) [reply testsuite/utils/pom.xml]: Not anymore

**Comment by mabartos** [comment]
@vmuzikar It should be in the final state right now :) Thanks for the feedback. I've analyzed all maven dependencies, and IMHO, this is the optimal solution.

**Review by vmuzikar — APPROVED** [review]
LGTM, thanks!

**Pull request merged**